lib/datasource/gitlab-tags/index.ts

```diff
@@ -1,136 +1,115 @@
-  if (cachedResult) {
-    return cachedResult;
+  constructor() {
+    super(GitlabTagsDatasource.id);
+    this.http = new GitlabHttp(GitlabTagsDatasource.id);
   }
 
-  const urlEncodedRepo = encodeURIComponent(repo);
-
-  // tag
-  const url = joinUrlParts(
-    depHost,
-    `api/v4/projects`,
-    urlEncodedRepo,
-    `repository/tags?per_page=100`
-  );
-
-  const gitlabTags = (
-    await gitlabApi.getJson<GitlabTag[]>(url, {
-      paginate: true,
-    })
-  ).body;
-
-  const dependency: ReleaseResult = {
-    sourceUrl: getSourceUrl(repo, registryUrl),
-    releases: null,
-  };
-  dependency.releases = gitlabTags.map(({ name, commit }) => ({
-    version: name,
-    gitRef: name,
-    releaseTimestamp: commit?.created_at,
-  }));
-
-  const cacheMinutes = 10;
-  await packageCache.set(
-    cacheNamespace,
-    getCacheKey(depHost, repo),
-    dependency,
-    cacheMinutes
-  );
-  return dependency;
-}
-
-/**
- * gitlab.getDigest
- *
- * This function will simply return the latest commit hash for the configured repository.
- */
-export async function getDigest(
-  { lookupName: repo, registryUrl }: Partial<DigestConfig>,
-  newValue?: string
-): Promise<string | null> {
-  const depHost = getDepHost(registryUrl);
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+
+  @cache({
+    namespace: `datasource-${GitlabTagsDatasource.id}`,
+    key: ({ registryUrl, lookupName }: GetReleasesConfig) =>
+      `${getDepHost(registryUrl)}:${lookupName}`,
+  })
+  async getReleases({
+    registryUrl,
+    lookupName: repo,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const depHost = getDepHost(registryUrl);
+
+    const urlEncodedRepo = encodeURIComponent(repo);
+
+    // tag
+    const url = joinUrlParts(
+      depHost,
+      `api/v4/projects`,
+      urlEncodedRepo,
+      `repository/tags?per_page=100`
+    );
 
-  const cachedResult = await packageCache.get<string>(
-    cacheNamespace,
-    getCacheKey(depHost, repo, 'commit')
-  );
-  // istanbul ignore if
-  if (cachedResult) {
-    return cachedResult;
+    const gitlabTags = (
+      await this.http.getJson<GitlabTag[]>(url, {
+        paginate: true,
+      })
+    ).body;
+
+    const dependency: ReleaseResult = {
+      sourceUrl: getSourceUrl(repo, registryUrl),
+      releases: null,
+    };
+    dependency.releases = gitlabTags.map(({ name, commit }) => ({
+      version: name,
+      gitRef: name,
+      releaseTimestamp: commit?.created_at,
+    }));
+
+    return dependency;
   }
 
-  const urlEncodedRepo = encodeURIComponent(repo);
-  let digest: string;
-
-  try {
-    if (newValue) {
-      const url = joinUrlParts(
-        depHost,
-        `api/v4/projects`,
-        urlEncodedRepo,
-        `repository/commits/`,
-        newValue
+  @cache({
+    namespace: `datasource-${GitlabTagsDatasource.id}-commit`,
+    key: ({ registryUrl, lookupName }: GetReleasesConfig) =>
+      `${getDepHost(registryUrl)}:${lookupName}`,
+  })
```

Comment:
nit: I'd prefer the decorator 'attached' to the function instead of before the comment. This is a style choice, so feel free to ignore.